core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -406,6 +406,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
   def getBoolean(key: String, defaultValue: Boolean): Boolean = {
     getOption(key).map(_.toBoolean).getOrElse(defaultValue)
   }
+  /** Get a parameter as a Option[Boolean] */
+  def getOptionBoolean(key: String): Option[Boolean] = {
```

Comment:
Yeah, I think it is better to throw exception for non matching type rather than silently revert to default value. Could be bug prone during operation.